Applications/web_portal/web_app/library/source/pages/account_page/risk_page/risk_parameters_view.tsx

```diff
@@ -8,13 +12,317 @@ interface Properties {
 
   /** Used to lookup currency names and symbols. */
   currencyDatabase: Nexus.CurrencyDatabase;
+
+  /** Called on an update of the currency. */
+  onCurrencyChange?: (currency: Nexus.Currency) => void;
+
+  /** Called on an update of the buying power. */
+  onBuyingPowerChange?: (value: Nexus.Money) => void;
+
+  /** Called on an update of the net loss. */
+  onNetLossChange?: (value: Nexus.Money) => void;
+
+  /** Called on an update of the transition time. */
+  onTransitionTimeChange?: (duration: Beam.Duration) => void;
+}
+
+interface State {
+  breakpoint: Breakpoint;
 }
 
-interface State {}
+interface FooterProperties {
+  isButtonDisabled: boolean;
+  isUserAdmin: boolean;
+  message: FOOTER_MESSAGE;
+  comment?: string;
+  onButtonClick: () => void;
+  onCommentChange: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
+}
+
+enum Breakpoint {
+  SMALL,
+  MEDIUM,
+  LARGE
+}
+
+enum FOOTER_MESSAGE {
+  SAVED,
+  SUBMITTED,
+  SERVER_ISSUE,
+  NONE
+}
+
+interface LabelProperties {
```

Comment:
Move this to where you declare the LabelComponent. As a general principle elements should be as tight as possible.